src/Microsoft.AspNet.OData.Shared/Query/Expressions/SelectExpandBinder.cs

```diff
@@ -21,77 +19,73 @@ namespace Microsoft.AspNet.OData.Query.Expressions
-        Justification = "Class coupling acceptable.")]
+    [SuppressMessage("Microsoft.Maintainability", "CA1506:AvoidExcessiveClassCoupling", Justification = "Class coupling acceptable.")]
     internal class SelectExpandBinder
     {
-        private SelectExpandQueryOption _selectExpandQuery;
         private ODataQueryContext _context;
         private IEdmModel _model;
         private ODataQuerySettings _settings;
         private string _modelID;
 
-        public SelectExpandBinder(ODataQuerySettings settings, SelectExpandQueryOption selectExpandQuery)
+        public SelectExpandBinder(ODataQuerySettings settings, ODataQueryContext context)
         {
             Contract.Assert(settings != null);
-            Contract.Assert(selectExpandQuery != null);
-            Contract.Assert(selectExpandQuery.Context != null);
-            Contract.Assert(selectExpandQuery.Context.Model != null);
+            Contract.Assert(context != null);
+            Contract.Assert(context.Model != null);
             Contract.Assert(settings.HandleNullPropagation != HandleNullPropagationOption.Default);
 
-            _selectExpandQuery = selectExpandQuery;
-            _context = selectExpandQuery.Context;
+            _context = context;
             _model = _context.Model;
             _modelID = ModelContainer.GetModelID(_model);
             _settings = settings;
         }
 
-        public static IQueryable Bind(IQueryable queryable, ODataQuerySettings settings,
-            SelectExpandQueryOption selectExpandQuery)
+        public static IQueryable Bind(IQueryable queryable, ODataQuerySettings settings, SelectExpandQueryOption selectExpandQuery)
         {
             Contract.Assert(queryable != null);
+            Contract.Assert(selectExpandQuery != null);
 
-            SelectExpandBinder binder = new SelectExpandBinder(settings, selectExpandQuery);
-            return binder.Bind(queryable);
+            SelectExpandBinder binder = new SelectExpandBinder(settings, selectExpandQuery.Context);
+            return binder.Bind(queryable, selectExpandQuery);
         }
 
-        public static object Bind(object entity, ODataQuerySettings settings,
-            SelectExpandQueryOption selectExpandQuery)
+        public static object Bind(object entity, ODataQuerySettings settings, SelectExpandQueryOption selectExpandQuery)
         {
             Contract.Assert(entity != null);
+            Contract.Assert(selectExpandQuery != null);
 
-            SelectExpandBinder binder = new SelectExpandBinder(settings, selectExpandQuery);
-            return binder.Bind(entity);
+            SelectExpandBinder binder = new SelectExpandBinder(settings, selectExpandQuery.Context);
+            return binder.Bind(entity, selectExpandQuery);
         }
 
-        private object Bind(object entity)
+        private object Bind(object entity, SelectExpandQueryOption selectExpandQuery)
```

Comment:
>object [](start = 16, length = 6)

should some of these private methods be made static, since you're now passing in the parameters, rather than using instance members?

Reply:
It seems i can't make it static. the methods will call class's instance methods.

---
In reply to: [355531434](https://github.com/OData/WebApi/pull/1941#discussion_r355531434) [](ancestors = 355531434)

Reply:
Ah; I missed the reference to ProjectElement in GetProjectionLambda, which does use instance variables (i.e., for settings).

---
In reply to: [355620119](https://github.com/OData/WebApi/pull/1941#discussion_r355620119) [](ancestors = 355620119,355531434)